src/io/parquet/read/deserialize/binary/utils.rs

```diff
@@ -36,10 +36,10 @@ impl<O: Offset> Pushable<usize> for Offsets<O> {
 
 impl<O: Offset> Binary<O> {
     #[inline]
-    pub fn with_capacity(capacity: usize) -> Self {
+    pub fn with_capacity(capacity: usize, values_capacity: usize) -> Self {
```

Comment:
Maybe we should have this signature:

```rust
pub fn with_capacity(capacity: usize, values_capacity: Option<usize>) -> Self {
```

and then 

```rust
let values_capacity = values_capacity.unwrap_or(capacity.min(100) * 24);
```

Reply:
If the values_capacity is zero by default, maybe we could use a `need_estimated` to reserve the space.

We use this in [databend](https://github.com/datafuselabs/databend/blob/8be55847084afb38f9c92d28ee10eb2f52bed90c/src/query/expression/src/types/string.rs#L318-L328)